build/linux/installer/scripts/tomlparser-health-config.rb

```diff
@@ -13,32 +13,83 @@
 @configMapMountPath = "/etc/config/settings/agent-settings"
 @configSchemaVersion = ""
 @enable_health_model = false
+@nodesChunkSize = 0
+@podsChunkSize = 0
+@eventsChunkSize = 0
+@deploymentsChunkSize = 0
+@hpaChunkSize = 0
+@podsEmitStreamBatchSize = 0
+@nodesEmitStreamBatchSize = 0
+
+def is_number?(value)
+  true if Integer(value) rescue false
+end
 
 # Use parser to parse the configmap toml file to a ruby structure
 def parseConfigMap
   begin
     # Check to see if config map is created
     if (File.file?(@configMapMountPath))
-      puts "config::configmap container-azm-ms-agentconfig for agent health settings mounted, parsing values"
+      puts "config::configmap container-azm-ms-agentconfig for agent settings mounted, parsing values"
       parsedConfig = Tomlrb.load_file(@configMapMountPath, symbolize_keys: true)
       puts "config::Successfully parsed mounted config map"
       return parsedConfig
     else
-      puts "config::configmap container-azm-ms-agentconfig for agent health settings not mounted, using defaults"
+      puts "config::configmap container-azm-ms-agentconfig for agent settings not mounted, using defaults"
       return nil
     end
   rescue => errorStr
-    ConfigParseErrorLogger.logError("Exception while parsing config map for enabling health: #{errorStr}, using defaults, please check config map for errors")
+    ConfigParseErrorLogger.logError("Exception while parsing config map for agent settings : #{errorStr}, using defaults, please check config map for errors")
     return nil
   end
 end
 
 # Use the ruby structure created after config parsing to set the right values to be used as environment variables
 def populateSettingValuesFromConfigMap(parsedConfig)
   begin
-    if !parsedConfig.nil? && !parsedConfig[:agent_settings].nil? && !parsedConfig[:agent_settings][:health_model].nil? && !parsedConfig[:agent_settings][:health_model][:enabled].nil?
+    if !parsedConfig.nil? && !parsedConfig[:agent_settings].nil?
+      if !parsedConfig[:agent_settings][:health_model].nil? && !parsedConfig[:agent_settings][:health_model][:enabled].nil?
         @enable_health_model = parsedConfig[:agent_settings][:health_model][:enabled]
         puts "enable_health_model = #{@enable_health_model}"
+      end
+      chunk_config = parsedConfig[:agent_settings][:chunk_config]
+      if !chunk_config.nil?
+        nodesChunkSize = chunk_config[:NODES_CHUNK_SIZE]
+        if !nodesChunkSize.nil? && is_number?(nodesChunkSize)
```

Comment:
check between a min & max, so one cannot ddos the api server by say setting this setting to 1.

Reply:
Sure, make sense. thanks for the feedback. Added min and max checks for both chunk and emit sizes.